app/javascript/packs/application.js

```diff
@@ -3,4 +3,36 @@ import 'mapbox-gl/dist/mapbox-gl.css'; // <-- you need to uncomment the styleshe
 
 import { initMapbox } from '../plugins/init_mapbox';
 
-initMapbox();
\ No newline at end of file
+import flatpickr from "flatpickr";
+
+
+initMapbox();
+
+// First we define two variables that are going to grab our inputs field. You can check the ids of the inputs with the Chrome inspector.
+const startDateInput = document.getElementById('rental_start_date');
+const endDateInput = document.getElementById('rental_end_date');
+
+// Check that the query selector id matches the one you put around your form.
+if (startDateInput) {
+const unavailableDates = JSON.parse(document.querySelector('#listing-booking-dates').dataset.unavailable)
+endDateInput.disabled = true
+console.log(endDateInput)
+
+flatpickr(startDateInput, {
+  minDate: "today",
+  disable: unavailableDates,
+  dateFormat: "Y-m-d",
+});
+
+
+startDateInput.addEventListener("change", (e) => {
+  if (startDateInput != "") {
+    endDateInput.disabled = false
+  }
+  flatpickr(endDateInput, {
+    minDate: e.target.value,
+    disable: unavailableDates,
+    dateFormat: "Y-m-d"
+    });
+  })
+};
```

Comment:
Please move all this code into a dedicated plugin .js file and import it into this application.js file, see how Luka has done it for the map JS for reference.